src/ripple/net/impl/RPCCall.cpp

```diff
@@ -660,25 +660,30 @@ class RPCParser
         {
             std::string const strParam = jvParams[i].asString ();
 
-            // Parameters 0 and 1 may be accounts
-            if ((i < 2) && (parseBase58<PublicKey> (
-                                TokenType::TOKEN_ACCOUNT_PUBLIC, strParam) ||
-                               parseBase58<AccountID> (strParam) ||
-                               parseGenericSeed (strParam)))
-            {
-                jvRequest[accFields[i]] = strParam;
+            if (i==1 && strParam.empty())
                 continue;
-            }
 
-            // Parameters 1 and 2 may be ledgers, but only if it's the last param
-            if (i > 0 && i == nParams - 1)
+            // Parameters 0 and 1 are accounts
+            if (i < 2)
+            {
+                if (parseBase58<PublicKey> (
+                        TokenType::TOKEN_ACCOUNT_PUBLIC, strParam) ||
+                    parseBase58<AccountID> (strParam) ||
+                    parseGenericSeed (strParam))
+                {
+                    jvRequest[accFields[i]] = strParam;
```

Comment:
Super Nitpick: Have `return parseAccountRaw2 (jvParams, "peer");` to use jss::peer